src/pages/index.js

```diff
@@ -60,40 +99,200 @@ const cardList = new Section(
+ *                               addNewCard                               *
+ **************************************************************************/
 const addNewCard = new PopupWithForm(
   "#profile-add-modal",
   handleAddCardFormSubmit
 );
 addNewCard.setEventListeners();
 
-//addCard .................................
+/**************************************************************************
+ *                               User Info                                *
+ **************************************************************************/
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__avatar"
+);
 
-//edit profile .................................
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUsersInfo(res);
+    userInfo.updateAvaImg(res);
+  })
+  .catch((err) => alert(err));
+
+/**************************************************************************
+ *                               Edit Profile                             *
+ **************************************************************************/
 
 const editProfileModal = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 editProfileModal.setEventListeners();
 
-//edit profile .................................
+function handleProfileEditSubmit(formValues) {
+  editProfileModal.setLoading(true);
+  api
+    .updateUserInfo({
+      title: formValues.name,
+      description: formValues.description,
+    })
+    .then(() => {
+      userInfo.setUsersInfo({
+        name: formValues.name,
+        about: formValues.description,
+      });
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error with updating Info.", err);
+    })
+    .finally(() => {
+      console.log("Profile Updated");
+      editProfileModal.setLoading(false);
+    });
+}
 
-//Popup with image..............................
+/**************************************************************************
+ *                               Edit Avatar                              *
+ **************************************************************************/
+const profileAvaForm = document.querySelector("#modal-ava-form");
+const profileAvaFormValidator = new FormValidator(
+  validationSettings,
+  profileAvaForm
+);
+profileAvaFormValidator.enableValidation();
 
-const imagePopup = new PopupWithImg("#modal-preview");
-imagePopup.setEventListeners();
+const newAvaImgModal = new PopupWithForm("#modal-ava", handleAvaEditSubmit);
+newAvaImgModal.setEventListeners();
 
-//Event Listeners ................................. //Event Handlers ................................. //Popup with image..............................
+function handleAvaEditSubmit(data) {
+  newAvaImgModal.setLoading(true);
+  api
+    .setUserAvatar(data.url)
+    .then((res) => {
+      userInfo.updateAvaImg(res);
+      newAvaImgModal.setEventListeners();
+    })
+    .catch((err) => {
+      console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors